app/javascript/components/enrollment/steps/Identification.js

```diff
@@ -152,36 +187,29 @@ class Identification extends React.Component {
           for (var issue of err.inner) {
             issues[issue['path']] = issue['errors'];
           }
-          self.setState({ errors: issues });
+          this.setState({ errors: issues });
         }
       });
   };
 
-  getLanguageOptions() {
-    const langOptions = supportedLanguages.languages.map(lang => {
-      const fullySupported = lang.supported.sms && lang.supported.email && lang.supported.phone;
-      const langLabel = fullySupported ? lang.name : lang.name + '*';
-      return { value: lang.name, label: langLabel };
-    });
-    return langOptions;
-  }
-
-  getLanguageValue = language => {
-    return this.state.languageOptions.find(lang => lang.value === language);
-  };
-
-  renderPrimaryLanguageSupportMessage(selectedLanguage) {
-    if (selectedLanguage) {
-      const languageJson = supportedLanguages.languages.find(l => l.name === selectedLanguage);
+  /**
+   * @param {Object} language - in the format of { c: isoCode, d: displayName }
+   * @returns {String} - the support message if the Primary Language has any limitations
+   */
+  getPrimaryLanguageMessage = language => {
```

Comment:
I feel like the language supported information should be stored with the major list and retrieved above like the display name is in `componentDidMount`.  Then this can be reverted to the render it was and `primaryLanguageMessage` does not need to be stored in state -  cleaner and less to track when updating the state

Reply:
changing that now!